File: src/com/personal/Fibonacci.java
package com.personal;

/**
 * Created by prajeev on 9/3/17.
 */
public class Fibonacci {

    public void printFibonacciNumbers(int iterations){
        int m=0;
        int n=1;
        System.out.println(n);
        for(int i = 0; i< iterations-1 ;i++){
            int sum = m+n;
            System.out.println(sum);
            m=n;
            n=sum;
        }

    }

    public void fibonacciRecursive(int howMany){
        fibonacciRecursiveInternal(0,1,howMany,0);
    }

    private void fibonacciRecursiveInternal(int m, int n ,int howMany, int counter) {
        if(counter == howMany){
            return;
        }else {
           // System.out.println(m);
            System.out.println(n);
            fibonacciRecursiveInternal(n, m + n, howMany, ++counter);
        }
    }

    public int printNthFibonacciNumber(int n) {
        if(n<2){
            return 1;
        } else {
            return printNthFibonacciNumber(n-1) + printNthFibonacciNumber(n-2);
        }
    }
}

File: src/com/functional/Fibonacci.scala
/**
  * Created by prajeeva on 9/22/17.
  */

object Fibonacci {
  def main(args: Array[String]): Unit = {
    for(i <- 1 to 10) println(fib(i))

    var m = 0
    var n = 1

    for(i <- 0 to 10 ) {
      val sum = m+n
      println(sum)
      m = n
      n = sum
    }
  }

  def fib(n: Int):Int = {
    if(n <= 1) n
    else {
      fib(n-2) + fib(n-1)
    }
  }
}

